### PR TITLE
Using common defaults in gcs and gcsio

### PR DESCRIPTION
**Pull request by abmodi**

### CODE REVIEW

**Review by arunkumarchacko — COMMENTED** [review]
Do we need any testing for this since this can have some ramifications. Also if you cant include some rationale behind changing some of the defaults, it would be helpful.

**Review by arunkumarchacko — APPROVED** [review]
Discussed offline with Abhishek. We are not changing the defaults, but making GCSIO defaults the same as GCS defaults.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java [hunk 1]
```diff
@@ -16,6 +16,28 @@
 
 package com.google.cloud.hadoop.fs.gcs;
 
+import static com.google.cloud.hadoop.gcsio.GoogleCloudStorageOptions.BATCH_THREADS_DEFAULT;
```
Comment (medb) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java:19]: Probably we need to converge on the default vars naming and prefix all of them with `DEFAULT_` prefix for consistency?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java [hunk 2]
```diff
@@ -326,11 +356,12 @@ public class GoogleHadoopFileSystemConfiguration {
 
   /** Configuration key for enabling use of the gRPC API for read/write. */
   public static final HadoopConfigurationProperty<Boolean> GCS_GRPC_ENABLE =
-      new HadoopConfigurationProperty<>("fs.gs.grpc.enable", false);
+      new HadoopConfigurationProperty<>("fs.gs.grpc.enable", ENABLE_GRPC_DEFAULT);
```
[the remark this answers is not in the document]
Comment (medb) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java:359]: +1

**Review by medb — CHANGES_REQUESTED** [review]

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java (hunk 1 above)
Comment (abmodi) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java:19]: All these are public constants previously defined. I will change for gRPC variables as it is not being.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java [hunk 3]
```diff
@@ -293,27 +320,30 @@ public class GoogleHadoopFileSystemConfiguration {
...
-   * If true, reading a file with GZIP content encoding (HTTP header "Content-Encoding: gzip") will
+   * If false, reading a file with GZIP content encoding (HTTP header "Content-Encoding: gzip") will
    * result in failure (IOException is thrown).
    */
   public static final HadoopConfigurationProperty<Boolean>
       GCS_INPUT_STREAM_SUPPORT_GZIP_ENCODING_ENABLE =
           new HadoopConfigurationProperty<>(
-              "fs.gs.inputstream.support.gzip.encoding.enable", false);
+              "fs.gs.inputstream.support.gzip.encoding.enable",
+              DEFAULT_GZIP_ENCODING_SUPPORT_ENABLED);
```
Comment (abmodi) [inline gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java:334]: @medb  could you please look into this - I have enabled gzip by default. Previously, it was disabled here but was enabled in gcsio. Does it look correct?

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java [hunk 4]
```diff
@@ -339,28 +370,33 @@ public class GoogleHadoopFileSystemConfiguration {
 
   /** Configuration key for check interval (in millisecond) for gRPC request timeout to GCS. */
   public static final HadoopConfigurationProperty<Long> GCS_GRPC_CHECK_INTERVAL_TIMEOUT_MS =
-      new HadoopConfigurationProperty<>("fs.gs.grpc.checkinterval.timeout.ms", 1_000L);
+      new HadoopConfigurationProperty<>(
+          "fs.gs.grpc.checkinterval.timeout.ms", GRPC_MESSAGE_TIMEOUT_CHECK_INTERVAL);
```
[the remark this answers is not in the document]
Comment (abmodi) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java]: Done.

File: gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java (hunk 3 above)
Comment (abmodi) [reply gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemConfiguration.java:334]: Disabled it by default now.

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageOptions.java [hunk 5]
```diff
@@ -35,10 +35,10 @@
...
   /** Default setting for enabling use of GCS gRPC API. */
-  public static final boolean ENABLE_GRPC_DEFAULT = false;
+  public static final boolean DEFAULT_GRPC_ENABLED = false;
 
   /** Default setting for enabling use of the Traffic Director for GCS gRPC API. */
-  public static final boolean ENABLE_TRAFFIC_DIRECTOR_DEFAULT = true;
+  public static final boolean DEFAULT_TRAFFIC_DIRECTOR_ENABLED = true;
 
   /** Default setting to prefer DirectPath for gRPC. */
   public static final boolean DIRECT_PATH_PREFERRED_DEFAULT = true;
```
Comment (medb) [inline gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageOptions.java:44]: Seems like we need to rename all default constants to have `DEFAULT_` prefix

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageMockitoTest.java [hunk 6]
```diff
@@ -77,7 +77,13 @@ public void setUp() {
     MockitoAnnotations.initMocks(this);
 
     MockHttpTransport transport = mockTransport();
-    gcs = mockedGcs(transport);
+    final GoogleCloudStorageOptions GCS_OPTIONS =
```
Comment (medb) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageMockitoTest.java:80]: We usually do not use final inside the methods, and all vars should be camelCase except class-level constants.

File: gcsio/src/test/java/com/google/cloud/hadoop/gcsio/testing/MockGoogleCloudStorageImplFactory.java [hunk 7]
```diff
@@ -48,4 +48,20 @@ public static GoogleCloudStorageImpl mockedGcs(HttpTransport transport) {
...
+        new Storage(
+            transport,
+            GsonFactory.getDefaultInstance(),
+            new TrackingHttpRequestInitializer(
+                new RetryHttpInitializer(
+                    new FakeCredentials(),
+                    RetryHttpInitializerOptions.builder()
+                        .setDefaultUserAgent("gcs-io-unit-test")
+                        .build()),
+                false));
```
Comment (medb) [inline gcsio/src/test/java/com/google/cloud/hadoop/gcsio/testing/MockGoogleCloudStorageImplFactory.java:64]: I think that this logic could be de-duplicated if you will call this new method inside the above `mockedGcs` method.

**Review by medb — APPROVED** [review]

File: gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageOptions.java (hunk 5 above)
Comment (abmodi) [reply gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageOptions.java]: Done. Thanks!

**Pull request merged**